src/main/scala/org/apache/spark/sql/pulsar/CachedPulsarClient.scala

```diff
@@ -62,7 +65,11 @@ private[pulsar] object CachedPulsarClient extends Logging {
       .removalListener(removalListener)
       .build[Seq[(String, Object)], Client](cacheLoader)
 
-  private def createPulsarClient(pulsarConf: ju.Map[String, Object]): Client = {
+  def createPulsarClient(
+                          pulsarConf: ju.Map[String, Object],
+                          pulsarClientBuilder: ClientBuilder = PulsarClient.builder()): Client = {
+    def hasConfig(key: String) = pulsarConf.get(key) != null
```

Comment:
Method `hasConfig` is unused. Plz check if you need it, otherwise just remove it.

Reply:
Removed it.